test_data/cloud-init/ubuntu/local/network-config

```diff
@@ -0,0 +1,5 @@
+version: 2
+ethernets:
+  ens4:
```

Comment:
How reliable is the net interface name?

Reply:
This is what I found across cloud-hypervisor docs and tests. So I thought it might have some reason for this already.
$ grep -inr "ens4"
tests/integration.rs:938:        let iface = "ens4";
tests/integration.rs:2496:        let iface = "ens4";
test_data/cloud-init/ubuntu/local/network-config:3:  ens4:
docs/live_migration.md:81:vm-1:~$ sudo ip addr add 192.168.101.2/24 dev ens4
docs/live_migration.md:82:vm-1:~$ sudo ip link set up dev ens4
docs/live_migration.md:113:vm-2:~$ sudo ip addr add 192.168.102.2/24 dev ens4
docs/live_migration.md:114:vm-2:~$ sudo ip link set up dev ens4
docs/custom-image.md:258:sudo ip addr add 192.168.2.10/24 dev ens4
docs/custom-image.md:259:sudo ip link set up dev ens4
docs/custom-image.md:261:sudo resolvectl dns ens4 8.8.8.8

Reply:
Yes, they are reliable in the context as the VMs are launched with specific configurations. So I guess the question will be, are we expecting users to use this cloud-init data and script only for examples listed in our README?

Reply:
default configuration in `local` folder mostly for `local testing (password and possibly a default network config that they can use to ease configuration.)` https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4773#issue-1406476575

Reply:
The naming of network interface (e.g. "ens4") will dependent a lot on how many other devices are already "in-use" on the guest VM. Since we don't have a control over users' VM configuration, the `network-config` content will not work for any "local testing".

IIUC, the local cloud-init and the script are only intended for being used together with our README where specific VM configurations are provided as examples. I think it is better to make this clear in the commit message, and also in the script/cloud-init data.

Reply:
The device is called ens4 due to the PCI slot number which can of course vary based on what the other devices are. Lets update the README to use a fixed MAC address and match on that instead,

Reply:
Modified as per comments, please check it.